mcp-spi/src/main/java/io/modelcontextprotocol/schema/McpSchemaCodec.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2024-2024 the original author or authors.
+ */
+
+package io.modelcontextprotocol.schema;
+
+import java.io.IOException;
+
+/**
+ * Codec interface for encoding and decoding JSON-RPC messages.
+ *
+ * @author Aliaksei Darafeyeu
+ */
+public interface McpSchemaCodec {
+
+	byte[] encode(McpSchema.JSONRPCMessage message);
+
+	String encodeAsString(Object message) throws IOException;
```

Comment:
This method is only ever used for `McpSchema.JSONRPCMessage`  and `McpError` instances. It may make more sense to change the signature to `String encodeAsString(McpSchea.JSONRPCMessage message)` (which would match decodeFromString) and add a separate method or overload for encoding `McpError` instances.

Reply:
Actually, on second thought, I'm not sure why this is used with McpError at all, maybe those lines could just avoid going trhough the codec.